compiler/rustc_ast/src/ast.rs

```diff
@@ -1950,7 +1944,7 @@ impl TyKind {
     }
 
     pub fn is_unit(&self) -> bool {
-        if let TyKind::Tup(ref tys) = *self { tys.is_empty() } else { false }
+        matches!(self, TyKind::Tup(tys) if tys.is_empty())
```

Comment:
I do realize the inconsistency in thinking this code is readable while the other `matches!` is not. I have no good answer for you.

Reply:
Wow, this is rare. Feels a bit extreme for me.

Reply:
I do get what you're saying, @oli-obk. In fact I was a bit uneasy about that other `matches!` myself but submitted it anyway for consistency and to see what the feedback is. This inconsistency probably means that guards in `matches!`are not by themselves problematic. Certainly something to be careful with though.